From 779c2354a3878b433438a1dcd85e82e051cc6269 Mon Sep 17 00:00:00 2001
From: Brandon Brnich <b-brnich@ti.com>
Date: Tue, 25 Apr 2023 17:15:11 -0500
Subject: [PATCH 414/508] vxe-vxd: decoder: Improve performance of h265 decoder

h265 decoder performance has been significantly worse than h264 decoder
due to latency in gstreamer format negotiation. Remove this latency by
aligning the pixel resolution earlier.

Signed-off-by: Brandon Brnich <b-brnich@ti.com>
Signed-off-by: Darren Etheridge <detheridge@ti.com>
---
 drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c b/drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c
index 3f7626669f74..53c68e896df1 100644
--- a/drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c
+++ b/drivers/media/platform/img/vxe-vxd/decoder/vxd_v4l2.c
@@ -1377,6 +1377,9 @@ static int vxd_dec_try_fmt(struct file *file, void *priv, struct v4l2_format *f)
 	unsigned int i = 0;
 	int ret = 0;
 
+	pix_mp->width = ALIGN(pix_mp->width, HW_ALIGN);
+	pix_mp->height = ALIGN(pix_mp->height, HW_ALIGN);
+
 	if (V4L2_TYPE_IS_OUTPUT(f->type)) {
 		fmt = find_format(f, IMG_DEC_FMT_TYPE_OUTPUT);
 		if (!fmt) {
-- 
2.41.0

